design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,147 @@
+        cart_id: ID! @doc(description: "Cart ID")
+        shipping_methods: [String!]! @doc(description: "Input for shipping method")
+    ): SetShippingMethodsOnNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+}
+
+type Customer {
+    negotiable_quote (
+        id: ID!): NegotiableQuote @doc(description: "Get negotiable quote of a customer")
+
+    negotiable_quotes(
+        page_size: Int = 20,
+        current_page: Int = 1,
+        filter: NegotiableQuoteFilterInput
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+
+    estimateShippingCostForNegotiableQuote(
+        address: NegotiableQuoteShippingAddressInput!
+    ): [ShippingMethodsOutput]! @doc(description:"Estimate shipping costs for a negotiable quote")
+
+    estimateShippingCostByAddressIdForNegotiableQuote(
+        address_id: String!
+    ): [ShippingMethodsOutput]! @doc(description:"Estimate shipping costs for a negotiable quote by Address ID")
+}
+
+type NegotiableQuote {
+    quote_id: ID!  @doc(description: "Negotiable quote ID")
+    items: [NegotiableQuoteItem] @doc(description: "Negotiable quote item")
+    attachements: [AttachmentContent] @doc(description: "Negotiable quote Attachements")
+    comments: [NegotiableQuoteComment] @doc(description: "Negotiable quote comments")
+    totals: TotalsOutput! @doc(description: "Negotiable quote totals output")
+    payment_info: NegotiableQuotePaymentInfo! @doc(description: "Negotiable quote payment info.")
+    available_shipping_methods: [ShippingMethodsOutput]! @doc(description: "Available shipping methods")
+}
+
+input NegotiableQuoteFilterInput {
+    ids: FilterEqualTypeInput @doc(description: "Filter Customer Negotiable quotes with an negotiable quote ID or list of negotiable quote IDs")
+    name: FilterMatchTypeInput @doc(description: "Filter by display name of the negotiable quote")
+}
+
+type NegotiableQuoteItem {
+    id: ID!
+    quantity: Float!
+    product: ProductInterface!
+}
+
+type AttachmentContent @doc(description: "Negotiable quote attachment file") {
+    base64_encoded_data: String!
+    mime_type: String!
+    name: String!
+}
+
+type NegotiableQuoteComment {
+    entity_id: ID!
+    parent_id: ID!
+    creator_type: Int
+    is_decline: Int
+    is_draft: Int
+    creator_id: ID!
+    comment: String
+    created_at: String
+    attachments: [NegotiableQuoteCommentAttachments]
+}
+
+type NegotiableQuoteCommentAttachments {
+    attachment_id: ID!
+    comment_id: ID
+    file_name: String
+    file_path: String
+    file_type: String
+}
+
+type NegotiableQuotePaymentInfo {
+    payment_methods: [PaymentMethodsOutput]! @doc(description: "All list of payment options and totals")
+    totals: TotalsOutput! @doc(description: "List of totals")
+}
+
+type PaymentMethodsOutput {
+    code: String @doc(description: "Payment method code")
+    title: String @doc(description: "Payment method title")
+}
+
+type TotalsOutput {
```

Comment:
It does not make sense to introduce custom totals type for the negotiable quote and still depend on cart specific types on the level of its fields. What is the reason not to use `CartPrices`? It seems identical now.
```suggestion
```